src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1609,14 +1609,13 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
 
             // Special-case the common cases of 1, 2, and 3 separators, with manual comparisons against each separator.
-            else if (separators.Length <= 3)
+            else if ((uint)separators.Length <= (uint)3)
```

Comment:
Is this cast still needed?
AFAIK JIT recognizes this now.

Reply:
@gfoidl do you mean we no longer need the pattern `if ((uint)index > (uint)array.Length)` that we have everywhere in the tree? If so we should have an issue to remove it.

Reply:
That's great! Glad it got fixed, I'll get rid of it